### PR TITLE
Adding new properties to Consumption Usage Details

### PR DESCRIPTION
**Pull request by ampravinr**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by ampravinr** [comment]
@sandeepnl @asarkar84 @ms-premp Please review Usage Details changes

**Review by asarkar84 — APPROVED** [review]
Looks good.

**Comment by annatisch** [comment]
Thanks @ampravinr!
I think there might be an omission in the readme.md, as looking at the clients generated from this PR - many operations have been removed - is that intentional?
e.g. Python:
https://github.com/Azure/azure-sdk-for-python/pull/2877/files

**Comment by ampravinr** [comment]
@annatisch Thanks for the review. That is not intentional. Found one issue from my side and working on fixing the same.

**Comment by sandeepnl** [comment]
If i remember correctly python auto-gen was added to handle cases where there were minor non-breaking chganges which can be done without any human intervention. 
@lmazuel Can you please confirm if removing python SDK auto-gen is okay?

**Comment by lmazuel** [comment]
Hi @sandeepnl 
No, auto-gen is global, can be massive breaking or details. It generates the state of your current Readme, so if the content is not correct in the Python PR, this likely means your Readme is not correct or something's wrong.
Disabling Python in the Readme because the generated content is not ok, it's like disabling the fire alarm when the building burns because it's making too much noise :)
TL;DR; Please put back the auto-gen, and let's discuss if the content is not ok for some reasons.

**Comment by ampravinr** [comment]
Sure on it now. thanks a lot.

**Comment by ampravinr** [comment]
@annatisch - I have made all the necessary changes. Could you please verify and let me know if any issues.

**Comment by annatisch** [comment]
Thanks @ampravinr - the diff is still showing the loss of some operations groups:
https://github.com/Azure/azure-sdk-for-python/pull/2877/files#diff-ab141b6de2e9066430c19456d8754671

Have these been removed or renamed?

**Comment by ampravinr** [comment]
@annatisch - I verified these changes and the latest changes are good to go. 

None of these operations are present in our existing "azure-sdk-for-python/azure-mgmt-consumption/azure/mgmt/consumption/models/__init__.py" file. Also these operations are suppose to belongs to other namespace (cost-management). I believe the last commits - https://github.com/Azure/azure-rest-api-specs/commit/7d850c00b7019e0062d764a91c3298eedb1513be#diff-ebc1cee9670996e7d7037ed33bd69b9a and https://github.com/Azure/azure-rest-api-specs/commit/30f184680a26ef230862ae38be373294d39e2f8d#diff-ebc1cee9670996e7d7037ed33bd69b9a might have caused this.

We are good with the current changes since those which are missing is not supposed to be in there. Let me know if any issues.

**Comment by annatisch** [comment]
Thanks for the confirmation @ampravinr.
There seems to be a number of model validation errors in the examples:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/402794446

Could you take a look?

**Comment by ampravinr** [comment]
On it now. Thanks.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json [hunk 1]
```diff
@@ -0,0 +1,3881 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/balances": {
+      "get": {
+        "tags": [
+          "Balances"
+        ],
+        "operationId": "GetBalancesByBillingAccount",
```
Comment (lmazuel) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: You cannot have at the same time:
- GetBalancesByBillingAccount_ByBillingPeriod
- GetBalancesByBillingAccount

Since the syntax XXX_YYY means for Autorest "Creates an operation group called XXX with operation YYY", while the syntax. So there is conflict, if GetBalancesByBillingAccount means functions or operation group.

Seeing the url, I would instinctively do:
- Balances_GetByBillingAccount
- Balances_GetByBillingPeriod

Comment (ampravinr) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: ok I got it. I will make the changes. Thanks a lot @lmazuel

**Comment by ampravinr** [comment]
@annatisch , @lmazuel, @sandeepnl  - I have made the necessary changes. Could you please review and let me know if any issue. I see errors in #10878.14 and #10878.15 which I think is not because of our changes. Please let me know if its something I need to fix. Thanks.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json [hunk 2]
```diff
@@ -0,0 +1,3906 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLink
```
Comment (lmazuel) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: You have both at the same time a model called ReservationRecommendations and an operation ID called ReservationRecommendations.
Usually, the operation is the plural form and the model the singular form: Operation ID *ReservationRecommendations* has operations that returns list of *ReservationRecommendation*

Is this true that this model "ReservationRecommendations" is actually ONE recommendation. If yes, please rename it "ReservationRecommendation" (and ReservationRecommendationProperties and ReservationRecommendationListResult)

Having the same string for Operation group ID and a model *might* (depending of the language) create troubles.

**Review by lmazuel — APPROVED** [review]
One final comment you may address if it makes sense for your service.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json (hunk 2 above)
Comment (ampravinr) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-06-30/consumption.json]: Sure, got it. Let me take a look and will change it accordingly. Thanks again.

**Comment by ampravinr** [comment]
@lmazuel - I verified the same and we are good here. ResourceGroupDimensions_List operation is moved from Consumption to another namespace\swagger. The same is moved to - "\azure-rest-api-specs\specification\cost-management\resource-manager\Microsoft.CostManagement\stable\2018-05-31\costmanagement.json" now.

**Review by lmazuel — APPROVED** [review]

**Comment by ampravinr** [comment]
@annatisch - I have made the required changes and not seeing any errors from my side. Please review and let me know if any issues or if any concerns.

**Comment by annatisch** [comment]
Thanks @ampravinr - a couple of things:
- Why did you rename `CostTags` to `CostTagsResult`? This model is actually passed in as a request body for the operation `CostTags_CreateOrUpdate` so calling it "result" is doesn't quite fit :)
- Please remove the final 's' from the ends of these operations:
    - `"Marketplaces_ListByEnrollmentAccounts"`
    - `"Marketplaces_ListForBillingPeriodByEnrollmentAccounts"`

- I would like to clarify the different between the operations:
    * `budgets_get`
    * `budgets_getByResourceGroupName`

   Does this mean that the latter operation will retrieve only the budget for the specified resource group? If so the naming is a bit confusing because the meaning is a bit different then `budgets_listByResourceGroup`. Maybe this could be better clarified by naming like so:

    * `budets_get`
    * `ResourceGroupBudgets_get` (and the same for the equivalent operations, like `ResourceGroupBudgets_delete`)

    Unless I've misunderstood what these APIs do?

Thanks!

Thanks @lmazuel for taking a look :)

**Comment by ampravinr** [comment]
@annatisch - please find below:

- changed CostTags to CostTagsResult because we have operations with CostTags like CostTags_Get and according to the errors as well as per @lmazuel that is not a good practice to have it like that.
I will update the operation id to CostTag and will revert the CostTagsResult to CostTags if that is acceptable.

- For the Budgets,  the later is specific to Resource group. We would like to keep the same naming since all our other operations also follow the same naming. For example - UsageDetails_ListByBillingAccount, UsageDetails_ListByDepartment etc. So we would like to keep the same here. Also this is something from the exsting version (03-31) and we don't have changes in name for these operations.

- Removing the 's' from ends of these operations "Marketplaces_ListByEnrollmentAccounts" and
"Marketplaces_ListForBillingPeriodByEnrollmentAccounts"

**Comment by ampravinr** [comment]
Hi @annatisch, I have made the above changes. Please review and let me know your thoughts. (I have reverted CostTagsResult to CostTag and the operations are now CostTags_*. Let me know if any issues)

**Review by annatisch — APPROVED** [review]

**Comment by ampravinr** [comment]
Thanks @annatisch , @lmazuel , @sandeepnl for all the comments and help.
